include/libtorrent/torrent_handle.hpp

```diff
@@ -1265,6 +1281,24 @@ namespace aux {
 		// client builds.
 		std::shared_ptr<torrent> native_handle() const;
 
+		// Returns true if the torrent is active in its session; returns false if
+		// it has been removed.
+		//
+		// A ``torrent_handle`` remains valid for some time after
+		// ``session::remove_torrent()`` is called. It will become invalid only
+		// after all libtorrent tasks (such as I/O tasks) release their references
+		// to the torrent. Until this happens, ``torrent_handle::is_valid()`` will
+		// return true, and other operations such as ``torrent_handle::status()``
+		// will succeed.
+		//
+		// ``torrent_handle::is_in_session()`` can be used to figure out if
+		// the call to ``session::remove_torrent()`` has been made. It returns true
+		// before ``torrent_removed_alert`` is posted, and false afterward.
```

Comment:
although this is correct, it might make the reader think that it should return false until I *receive* `torrent_removed_alert`. I think specifying that it will return false until the call to `session::remove_torrent()` is made is simpler and just as correct. Since `is_in_session()` still is synchronous, it will always synchronize with a previous call to remove the torrent